### ISSUE TITLE
where imagetools?

### ISSUE THREAD

**Issue by wangbiao0** [markemus/wsireg#1]
Hi, I'm studying your code.
May I ask where can I find imagetools (in-house image processing package) in your code file in dependencies.txt?

**Comment by markemus** [comment]
Unfortunately I don't have access to it at the moment- it was a package I wrote which held a number of functions for reuse. It doesn't seem to be heavily used in this package though, you should be able to work around it. I saw that I used image.od, which is an optical density transformation of the RGB image, and ci.scale_by_max(img), which was a rescaling of the image by setting its highest value to 255 and rescaling the values accordingly (Macenko normalization).

However just to let you know- the registration function in this package is my own idea and not a mainstream method, and you're probably better off using a standard registration package like Elastix. You can use a b-spline transform in Elastix and get similarly good results for tissue. This approach could work better for more complex registrations like MRI scans but I only tested it on WSIs. May I ask what you're planning to use it for?

**Comment by wangbiao0** [comment]
Thanks for your reply, now I know how to modify the code.
I'm working on how to stitch microscope scans that need to be aligned before stitching.
But there is a big gap between natural and WSI images, which is a new area for me, and I would like to refer to some codes related to WSI alignment.

**Comment by markemus** [comment]
Then you should definitely try b-splines with Elastix, which was the most effective approach I found. The approach in this library is experimental. However if you still want to try it out let me know how it goes!

**Comment by wangbiao0** [comment]
Yeah, I'll try that. If I use your code and it works, I'll let you know the delightful news!

**Comment by markemus** [comment]
Great, good luck!

**Comment by markemus** [comment]
Great, good luck!
